app/src/main/java/com/android/shelfLife/ui/utils/UIComponents.kt

```diff
@@ -93,37 +95,62 @@ fun CustomTopAppBar(
 fun CustomSearchBar(
     query: String,
     onQueryChange: (String) -> Unit,
+    onDeleteTextClicked: () -> Unit,
     placeholder: String,
     searchBarTestTag: String
 ) {
   Box(
       modifier = Modifier.fillMaxWidth().padding(16.dp), // Outer padding for spacing
       contentAlignment = Alignment.Center // Center the SearchBar within the Box
       ) {
-        androidx.compose.material3.SearchBar(
-            colors =
-                SearchBarDefaults.colors(
-                    containerColor = MaterialTheme.colorScheme.surfaceContainerLowest,
-                ),
-            shadowElevation = 3.dp,
-            query = query,
-            onQueryChange = onQueryChange,
-            placeholder = { Text(placeholder) },
-            onSearch = {},
-            active = false,
-            onActiveChange = {},
-            leadingIcon = {},
-            trailingIcon = {
-              IconButton(onClick = {}) {
-                Icon(Icons.Default.Search, contentDescription = "Search Icon")
-              }
+        val colors1 =
+            SearchBarDefaults.colors(
+                containerColor = MaterialTheme.colorScheme.surfaceContainerLowest,
+            )
+        SearchBar(
+            inputField = {
+              SearchBarDefaults.InputField(
```

Comment:
I really like this InputField. It helps make the code more readable and separates the components inside SearchBar nicely